plugin-server/src/cdp/templates/_destinations/native-webhook/webhook.template.ts

```diff
@@ -6,86 +6,105 @@ export const template: NativeTemplate = {
     type: 'destination',
     id: 'native-webhook',
     name: 'Native HTTP Webhook',
-    description: 'Sends a native webhook templated by the incoming event data',
-    icon_url: '/static/posthog-icon.svg',
-    category: ['Custom'],
-    perform: (request, { payload }) => {
-        try {
-            return request(payload.url, {
-                method: payload.method,
-                headers: payload.headers,
-                json: payload.body,
-            })
-        } catch (error) {
-            throw new Error(error.message)
-        }
-    },
+    description: 'Sends events to a native HTTP webhook',
+    icon_url: '/static/webhook-icon.svg',
+    category: ['Analytics'],
     inputs_schema: [
         {
-            key: 'url',
+            key: 'apiKey',
             type: 'string',
-            label: 'Webhook URL',
+            label: 'API Key',
             secret: false,
             required: true,
-            description: 'Endpoint URL to send event data to.',
+            description: 'API Key to authenticate the request.',
         },
+    ],
+    mapping_templates: [
         {
-            key: 'method',
-            type: 'choice',
-            label: 'Method',
-            secret: false,
-            choices: [
-                {
-                    label: 'POST',
-                    value: 'POST',
-                },
+            name: 'Send a request',
+            include_by_default: true,
+            filters: {
+                events: [
+                    {
+                        id: null,
+                        name: 'All events',
+                        order: 0,
+                        type: 'events',
+                    },
+                ],
+            },
+            associated_action: 'send',
```

Comment:
This feels like essentially a "new" feature that I feel has a bunch of issues tbh. 

We talked about this a bit in person but I feel strongly that this concept isn't proved out as a strong idea.
The examples given in this aren't doing anything special and could be completely replaced by just a simple input. 

Especially as there are some changes we need to consider around mappings that we can discuss separately